docs/quickstarts/react.mdx

```diff
@@ -41,216 +92,79 @@ Below is an example of an .env file. If you are signed into your Clerk Dashboard
 
-```tsx filename="app.tsx"
-import React from "react";
-import "./App.css";
-import { ClerkProvider } from "@clerk/clerk-react";
 
-if (!process.env.REACT_APP_CLERK_PUBLISHABLE_KEY) {
+```tsx filename="src/main.tsx" {5, 7, 9-11, 15, 17}
+import React from 'react'
+import ReactDOM from 'react-dom/client'
+import App from './App.tsx'
+import './index.css'
+import { ClerkProvider } from '@clerk/clerk-react'
+
+const PUBLISHABLE_KEY = import.meta.env.VITE_CLERK_PUBLISHABLE_KEY
+
+if (!PUBLISHABLE_KEY) {
   throw new Error("Missing Publishable Key")
 }
-const clerkPubKey = process.env.REACT_APP_CLERK_PUBLISHABLE_KEY;
 
-function App() {
-  return (
-    <ClerkProvider publishableKey={clerkPubKey}>
-      <div>Hello from clerk</div>
+ReactDOM.createRoot(document.getElementById('root')!).render(
+  <React.StrictMode>
+    <ClerkProvider publishableKey={PUBLISHABLE_KEY}>
+      <App />
     </ClerkProvider>
-  );
-}
-
-export default App;
+  </React.StrictMode>,
+)
 ```
+### Embed the `<UserButton />` and `<SignInButton />`
 
-### Protecting your pages
-
-Clerk offers [Control Components](/docs/components/overview) that allow you to protect your pages. These components are used to control the visibility of your pages based on the user's authentication state.
+Clerk offers a set of [prebuilt components](/docs/components/overview) to add functionality to your app with minimal effort. The [`<UserButton />`](/docs/components/user/user-button) allows users to manage their account information and log out, completing the full authentication circle.The [`<SignInButton />`](/docs/components/unstyled/sign-in-button) is an unstyled component that you can use to either create a basic button or to wrap around your styled button and turn it into a Sign In button.
 
-```tsx filename="app.tsx"
-import React from "react";
-import "./App.css";
-import {
-  ClerkProvider,
-  SignedIn,
-  SignedOut,
-  UserButton,
-  useUser,
-  RedirectToSignIn,
-} from "@clerk/clerk-react";
+Lastly you will use the [`<SignedIn />`](/docs/components/control/signed-in) and [`<SignedOut />`](/docs/components/control/signed-out) control components. These will allow you to render the `<UserButton />` only to users who are signed in and `<SignInButton />` only to users who are signed out.
 
-if (!process.env.REACT_APP_CLERK_PUBLISHABLE_KEY) {
-  throw "Missing Publishable Key"
-}
+Add the following anywhere inside `<ClerkProvider />` in your app. First add `import { UserButton, SignInButton, SignedIn, SignedOut } from "@clerk/clerk-react";` to the top of your file. Then add the following somewhere inside of your application -- a header area is a great place. The `afterSignOutUrl` prop lets you customize what page the user will be redirected to after sign out.
```

Comment:
You mention a header area, but the code example doesn't have anything about a header area. The code is simply embedded in a div in the App()

Reply:
Its a reference to a common practice. I can remove it though.  I'll trim that down.